core/src/main/scala/org/apache/spark/metrics/sink/opentelemetry/OpenTelemetryPushSink.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.metrics.sink.opentelemetry
+
+import java.util.{Locale, Properties}
+import java.util.concurrent.TimeUnit
+
+import com.codahale.metrics.MetricRegistry
+import org.apache.commons.lang3.StringUtils
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.metrics.sink.Sink
+
+private[spark] object OpenTelemetryPushSink {
+  private def fetchMapFromProperties(
+      properties: Properties,
+      keyPrefix: String): Map[String, String] = {
+    val propertiesMap = scala.collection.mutable.Map[String, String]()
+    val valueEnumeration = properties.propertyNames
+    val dotCount = keyPrefix.count(_ == '.')
+    while (valueEnumeration.hasMoreElements) {
+      val key = valueEnumeration.nextElement.asInstanceOf[String]
+      if (key.startsWith(keyPrefix)) {
+        val dotIndex = StringUtils.ordinalIndexOf(key, ".", dotCount + 1)
+        val mapKey = key.substring(dotIndex + 1)
+        propertiesMap(mapKey) = properties.getProperty(key)
+      }
+    }
+    propertiesMap.toMap
+  }
+}
+
+private[spark] class OpenTelemetryPushSink(val property: Properties, val registry: MetricRegistry)
+    extends Sink
```

Comment:
2-space indentation.

Reply:
after running `./dev/scalafmt`, the file is auto recorrected to four spaces.

Reply:
Changed in https://github.com/apache/spark/pull/47763/commits/b4ae2df0d9bd4d75df54a53f596d6b6045c8c354